frontend/src/components/utils/autoRefreshHandler.tsx

```diff
@@ -119,18 +121,43 @@ export default (
     };
 
     render() {
-      return this.state.display ? (
-        <InfiniteScroll
-          dataLength={this.state.items.length}
-          next={this.fetchMoreData}
-          hasMore={this.state.hasMore}
-          loader={<p>Loading</p>}
-          style={{ overflowX: "hidden" }}
-        >
-          <WrappedComponent items={this.state.items} {...props} />
-        </InfiniteScroll>
-      ) : (
-        <PaginationSpinner hidden={false} />
+      if (!this.state.display) {
+        return <PaginationSpinner hidden={false} />;
+      }
+      return (
+        <>
+          <InfiniteScroll
+            dataLength={this.state.items.length}
+            next={this.fetchMoreData}
+            hasMore={this.state.hasMore}
+            loader={<PaginationSpinner hidden={false} />}
+            style={{ overflowX: "hidden" }}
+          >
+            <WrappedComponent items={this.state.items} {...props} />
+            {this.state.hasMore && (
```

Comment:
1. Hide the button when the data is loading
2. Let's re-use the "VIEW ALL" button design from the dashboard here and put it in the middle